### PR TITLE
Initial Documentation + API dostrings

### PR DESCRIPTION
**Pull request by odunbar**
## Purpose
Establish initial docs pages + pages of doc-strings tackling a piece of #1.

## In this PR
- Basic pages: Home, Contributor instructions, Installation
- Docstrings for all src/ and API pages for all src/ - based on the `DocStringsExtensions` `TYPEDXYZ` keywords
- Add GH workflow (from EKP) to create documentation previews for PRs
## Also...
- Added a parametric type to `Decomposition`

### CODE REVIEW

File: docs/src/installation_instructions.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Installation
+
+RandomFeatures.jl is a registered Julia package. You can install the latest version
```
Comment (tsj5) [inline docs/src/installation_instructions.md]: Not a registered package yet, right? Think only instructions should be to clone the git repo. Unfortunately I lost my version of installation_instructions.md when I rebased #3 .

File: docs/src/index.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# RandomFeatures
+
+A julia package to construct and apply random feature methods for regression.
+RandomFeatures can be viewed as an approximation of kernel methods. They can be used both as a substitution in Kernel ridge regression and Gaussian Process regresison. 
+
+
+## Highlights
```
Comment (tsj5) [inline docs/src/index.md:15]: Can skip this on a first pass, but think index.md conventionally has a table listing modules and a one-line summary of their purpose (along the lines of the walkthrough you gave me on Tues).

**Review by tsj5 — CHANGES_REQUESTED** [review]
Looks good! All standard stuff -- only comment is to edit installation_instructions.md since the package hasn't been registered yet.

File: docs/src/installation_instructions.md (hunk 1 above)
Comment (odunbar) [reply docs/src/installation_instructions.md]: Done

File: docs/src/index.md (hunk 2 above)
Comment (odunbar) [reply docs/src/index.md:15]: Done anyway, it only took a minute I've kept it brief though until things are set!

**Comment by odunbar** [comment]
merging as changes have been addressed...

**Pull request merged**